drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c

```diff
@@ -1004,32 +1004,25 @@ int wifi_nrf_wpa_supp_signal_poll(void *if_priv, struct wpa_signal_info *si, uns
 	fmac_dev_ctx = rpu_ctx_zep->rpu_ctx;
 
 	vif_ctx_zep->signal_info = si;
-	ret = wifi_nrf_fmac_get_station(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx, bssid);
-	if (ret != WIFI_NRF_STATUS_SUCCESS) {
-		LOG_ERR("%s: Failed to get station info\n", __func__);
-		goto out;
-	}
 
-	sem_ret = k_sem_take(&wait_for_event_sem, K_MSEC(RPU_RESP_EVENT_TIMEOUT));
-	if (sem_ret) {
-		LOG_ERR("%s: Failed to get station info, ret = %d\n", __func__, sem_ret);
-		ret = WIFI_NRF_STATUS_FAIL;
-		goto out;
-	}
-
-	ret = wifi_nrf_fmac_get_interface(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx);
-	if (ret != WIFI_NRF_STATUS_SUCCESS) {
-		LOG_ERR("%s: Failed to get interface info\n", __func__);
-		goto out;
-	}
+	if ((wifi_nrf_osal_time_elapsed_us(fmac_dev_ctx->fpriv->opriv,
+		vif_ctx_zep->rssi_record_timestamp_us) / 1000) > MAX_RSSI_RECORD_TIME_GAP_MS) {
```

Comment:
Shouldn't this be reverse? If the time elapsed is > MAX_RSSI_RECORD_TIME_GAP_MS then its time for us to query RPU

Reply:
Yes it should be